editor/import/scene_import_settings.cpp

```diff
@@ -406,6 +406,134 @@ void SceneImportSettings::_fill_scene(Node *p_node, TreeItem *p_parent_item) {
+				Vector3 v1 = skeleton->get_bone_global_rest(child_bone_idx).origin;
+				Vector3 d = (v1 - v0).normalized();
+				real_t dist = v0.distance_to(v1);
+
+				// Find closest axis.
+				// int closest = -1;
+				// real_t closest_d = 0.0;
+				// for (int j = 0; j < 3; j++) {
+				// 	real_t dp = Math::abs(skeleton->get_bone_global_rest(current_bone_idx).basis[j].normalized().dot(d));
+				// 	if (j == 0 || dp > closest_d) {
+				// 		closest = j;
+				// 	}
+				// }
+
+				// Draw bone.
+				switch (bone_shape) {
+					case 0: { // Wire shape.
+						surface_tool->set_color(bone_color);
+						bones[0] = current_bone_idx;
+						surface_tool->set_bones(bones);
+						surface_tool->set_weights(weights);
+						surface_tool->add_vertex(v0);
+						bones[0] = child_bone_idx;
+						surface_tool->set_bones(bones);
+						surface_tool->set_weights(weights);
+						surface_tool->add_vertex(v1);
+					} break;
+
+					case 1: { // Octahedron shape.
+						Vector3 first;
+						Vector3 points[6];
+						int point_idx = 0;
+						for (int j = 0; j < 3; j++) {
+							Vector3 axis;
+							if (first == Vector3()) {
+								axis = d.cross(d.cross(skeleton->get_bone_global_rest(current_bone_idx).basis[j])).normalized();
+								first = axis;
+							} else {
+								axis = d.cross(first).normalized();
+							}
+
+							surface_tool->set_color(bone_color);
+							for (int k = 0; k < 2; k++) {
+								if (k == 1) {
+									axis = -axis;
+								}
+								Vector3 point = v0 + d * dist * 0.2;
+								point += axis * dist * 0.1;
+
+								bones[0] = current_bone_idx;
+								surface_tool->set_bones(bones);
+								surface_tool->set_weights(weights);
+								surface_tool->add_vertex(v0);
+								surface_tool->set_bones(bones);
+								surface_tool->set_weights(weights);
+								surface_tool->add_vertex(point);
+
+								surface_tool->set_bones(bones);
+								surface_tool->set_weights(weights);
+								surface_tool->add_vertex(point);
+								bones[0] = child_bone_idx;
+								surface_tool->set_bones(bones);
+								surface_tool->set_weights(weights);
+								surface_tool->add_vertex(v1);
+								points[point_idx++] = point;
+							}
+						}
+						surface_tool->set_color(bone_color);
+						SWAP(points[1], points[2]);
+						bones[0] = current_bone_idx;
+						for (int j = 0; j < 6; j++) {
+							surface_tool->set_bones(bones);
+							surface_tool->set_weights(weights);
+							surface_tool->add_vertex(points[j]);
+							surface_tool->set_bones(bones);
+							surface_tool->set_weights(weights);
+							surface_tool->add_vertex(points[(j + 1) % 6]);
+						}
+					} break;
+				}
+
+				// Add the bone's children to the list of bones to be processed.
+				bones_to_process.push_back(child_bones_vector[i]);
+			}
+		}
+
+		Ref<ArrayMesh> m = surface_tool->commit();
+		m->surface_set_material(0, bones_preview_mesh_material);
+		bones_preview_mesh_instance->set_mesh(m);
+	}
```

Comment:
Resend from: https://github.com/godotengine/godot-proposals/issues/6801#issuecomment-1538702093

> We have discussed about this and review https://github.com/godotengine/godot/pull/76818 with @lyuma, the main problem is that there is a lot of duplication of code from the SkeletonGizmo.

Reply:
I'm thinking maybe a static function that returns the mesh representation of the bones (with the material already applied). Would that be ok?

Reply:
Ah, yes. It might not be a great architectural design, but it certainly might be possible to avoid code duplication with static functions.

Probably I would not block that, but we need to hear the decisions of other contributors as well.

Reply:
I think static functions is better than nothing, but I have no opinion.

Reply:
I think we all agree static functions are awful 😁

Is there other way we could do this?

Reply:
Well, it seems that no one is particularly opposed to that way, so for now I think it is fine to proceed with the implementation as static function.

Reply:
Are all the reviews ready?